astropy/units/utils.py

```diff
@@ -174,18 +174,22 @@ def sanitize_scale(scale):
     if is_effectively_unity(scale):
         return 1.0
 
-    if np.iscomplex(scale):  # scale is complex
-        if scale == 0.0:
-            return 0.0
+    # Maximum speed for regular case where scale is a float.
+    if scale.__class__ is float:
+        return scale
 
+    if scale.imag:
```

Comment:
I was surprised that it's safe to do this after only checking `is float`, but I learned in checking that both `int` and `Fraction()` both have the `.imag` attribute!

Reply:
TIL:

```
>>> (1).imag
0
```